app/models/post.rb

```diff
@@ -1,17 +1,17 @@
 class Post < ApplicationRecord
   belongs_to :author, class_name: 'User'
-  has_many :comments, foreign_key: :comment_id
-  has_many :likes, foreign_key: :post_id
+  has_many :comments
+  has_many :likes
 
   validates :title, presence: true, length: { maximum: 250 }
   validates :comments_counter, numericality: { integer: true, greater_than_or_equal_to: 0 }
   validates :likes_counter, numericality: { integer: true, greater_than_or_equal_to: 0 }
 
   def update_users_posts_counter
-    user.increment!(:post_counter)
+    @user.increment!(:post_counter)
   end
 
```

Comment:
- [ ] Your issue related to the post counter not being updated has to do with this method. It is not defined correctly. A post belongs to an `author`, not a `@user`.

  I would also recommend you add the `after_save` callback so that this method is triggered automatically.

  ```
  has_many :likes
  after_save :update_users_posts_counter

  ...

  def update_users_posts_counter
    author.increment!(:post_counter)
  end
  ```